tf_patches/gpu_race_condition.diff

```diff
@@ -0,0 +1,14 @@
+diff --git a/tensorflow/compiler/xla/service/gpu/gpu_executable.cc b/tensorflow/compiler/xla/service/gpu/gpu_executable.cc
+index ed55b32c47a..260fddb3f5c 100644
+--- a/tensorflow/compiler/xla/service/gpu/gpu_executable.cc
++++ b/tensorflow/compiler/xla/service/gpu/gpu_executable.cc
+@@ -489,8 +489,7 @@ StatusOr<ExecutionOutput> GpuExecutable::ExecuteAsyncOnStreamImpl(
+       "GpuExecutable::ExecuteAsyncOnStreamImpl(", module_name_, ")"));
+   se::DeviceMemoryAllocator* const memory_allocator = run_options->allocator();
+   // Force synchronous execution if the allocator requires it.
+-  const bool block_host_until_done =
+-      !memory_allocator->AllowsAsynchronousDeallocation();
++  const bool block_host_until_done = true;
```

Comment:
Will this has speed implication?

Reply:
Appears not. For all the models I have tested: resnet50, ViT, bert, the speed remains the same with this patch.